src/models.py

```diff
@@ -193,6 +193,19 @@ def forward(self, x):
         return out
 
 
+class GoogleResNet50withPCEN(PCENMixin, GoogleResNet50):
```

Comment:
Not sure what's going on here. 
Double inheritance is a devil, do we really need it? can we just add self.pcen and apply it before the forward of GoogleResNet50? 
And I don't like the code of pcen, I think we should do a small refactor there (nothing to do with this pr though)

Reply:
what's the problem? The PCENMixin is really elegant in my opinion, have a look at it. Basically it adds a (possibly trainable) PCEN layer before the original model in the forward method
About PCEN code - well, let's have a new PR

Reply:
I don't have a problem with PCENMixin, I have a problem with the double inheritance, I don't think there is a justification to it.
the PCEN should be an attribute of GoogleResNet50withPCEN imo
let's zoom it